packages/connectors/connector-firebase/src/index.ts

```diff
@@ -0,0 +1,168 @@
+import { OpenloginAdapter } from "@web3auth/openlogin-adapter"
+import type { ChainNamespaceType } from "@web3auth/base"
+import type { UserCredential } from "@firebase/auth"
+import {
+	GoogleAuthProvider,
+	getAuth,
+	signInWithPopup,
+	getIdToken,
+} from "@firebase/auth"
+import { initializeApp } from "@firebase/app"
+
+const PROVIDER_ID = "firebase" as const
+
+export class FirebaseConnectionProvider extends AbstractConnectionProvider<
+  typeof PROVIDER_ID,
+EthereumProviderConnectionResult
+> {
+  private readonly instance: Observable<Web3AuthNoModal>;
+  private readonly connection: Observable<
+  ConnectionState<EthereumProviderConnectionResult>
+  >;
+
+  constructor(
+  	private readonly clientId: string,
+  	private readonly chainConfig: {
+  		chainNamespace: ChainNamespaceType;
+  		chainId: string;
+  		rpcTarget: string;
+  		displayName: string;
+  		blockExplorer: string;
+  		ticker: string;
+  		tickerName: string;
+  	},
+  	private readonly firebaseConfig: {
+  		apiKey: string;
+  		authDomain: string;
+  		projectId: string;
+  		storageBucket: string;
+  		messagingSenderId: string;
+  		appId: string;
+  	},
+  	private readonly network: OPENLOGIN_NETWORK_TYPE | undefined
+  ) {
+  	super()
+  	this.instance = cache(() => this._connect())
+  	this.connection = this.instance.pipe(
+  		mergeMap((instance) => {
+  			return connectToWeb3(instance.provider, {
+  				disconnect: this.disconnect.bind(this),
+  			})
+  		}),
+  		startWith(getStateConnecting({ providerId: PROVIDER_ID }))
+  	)
+  }
+
+  private async _connect(): Promise<Web3AuthNoModal> {
+  	const web3auth = new Web3AuthNoModal({
+  		clientId: this.clientId,
+  		web3AuthNetwork: this.network,
+  		chainConfig: this.chainConfig,
+  	})
+  	debugger
+
+  	const privateKeyProvider = new EthereumPrivateKeyProvider({
+  		config: { chainConfig: this.chainConfig },
+  	})
+
+  	const openloginAdapter = new OpenloginAdapter({
+  		adapterSettings: {
+  			uxMode: "popup",
+  			loginConfig: {
+  				jwt: {
+  					verifier: "firebase-dog-dev",
+  					typeOfLogin: "jwt",
+  					clientId: this.clientId,
+  				},
+  			},
+  		},
+  		privateKeyProvider,
+  	})
+
+  	web3auth.configureAdapter(openloginAdapter)
+
+  	await web3auth.init()
+  	if (web3auth.connected) return web3auth
+  	const app = initializeApp(this.firebaseConfig)
+  	const auth = getAuth(app)
+  	const provider = new GoogleAuthProvider()
+  	const userCredential: UserCredential = await signInWithPopup(
+  		auth,
+  		provider
+  	)
+  	try {
+  		const idToken = await getIdToken(userCredential.user)
+  		console.log("Firebase ID token", idToken)
+
+  		const web3authProvider = await web3auth.connectTo(
+  			WALLET_ADAPTERS.OPENLOGIN,
+  			{
+  				loginProvider: "jwt",
+  				mfaLevel: "none",
+  				extraLoginOptions: {
+  					id_token: idToken,
+  					verifierIdField: "sub",
+  					domain: "http://localhost:3000",
```

Comment:
I guess this parameter should be as part of config. Also check other parameters in your code